### DISCUSSION TITLE
Can't use LogisticRegression with SmartExplainer

### DISCUSSION THREAD

**Discussion by ciscinho** [MAIF/shapash#474] [Q&A]
Hi all,

I am  a new user of shapash.
I've tried to use SmartExplainer with sklearn LogisticRegression model but I've got an error when instantiating the object.

My code is very simple:
```
regressor = LogisticRegression().fit(X_train, y_train)
xpl = SmartExplainer(model=regressor)

```
and the error:
```
/opt/conda/envs/Fastpro/lib/python3.7/site-packages/shapash/backend/shap_backend.py in __init__(self, model, preprocessing, explainer_args, explainer_compute_args)
     16         self.explainer_args = explainer_args if explainer_args else {}
     17         self.explainer_compute_args = explainer_compute_args if explainer_compute_args else {}
---> 18         self.explainer = shap.Explainer(model=model, **self.explainer_args)
     19 
     20     def run_explainer(self, x: pd.DataFrame) -> dict:

/opt/conda/envs/Fastpro/lib/python3.7/site-packages/shap/explainers/_explainer.py in __init__(self, model, masker, link, algorithm, output_names, feature_names, linearize_link, seed, **kwargs)
    169                 # if we get here then we don't know how to handle what was given to us
    170                 else:
--> 171                     raise TypeError("The passed model is not callable and cannot be analyzed directly with the given masker! Model: " + str(model))
    172 
    173             # build the right subclass

TypeError: The passed model is not callable and cannot be analyzed directly with the given masker! Model: LinearRegression()
```

Is LogisticRegression compatible with shapash or I am missing something when instantiating ?

Thanks for your help

**Comment by ThomasBouche** [comment]
Hi, 

Shapash should work with logisticregression. We don't use it with Shapash and there must be a problem with the unit tests for the error not to show up. I'll create an issue. #475 

However, you can use it like that : 

Compute shap contributions: (https://shap.github.io/shap/notebooks/linear_explainer/Sentiment%20Analysis%20with%20Logistic%20Regression.html)

And then add contributions to Shapash:
https://github.com/MAIF/shapash/blob/master/tutorial/explainer_and_backend/tuto-expl05-Shapash-using-Fasttreeshap.ipynb

Currently, you still need to give a model that is works with Shapash, so you can give a fake randomforest model, for example.
It's a use of Shapash that needs to be reviewed, and it's on our roadmap.
